docs/application/native/guides/ui/dali/progressbar-n.md

```diff
@@ -0,0 +1,120 @@
+progressBar.SetAnchorPoint(AnchorPoint::TOP_CENTER);
+progressBar.SetResizePolicy(ResizePolicy::FILL_TO_PARENT, Dimension::WIDTH);
+progressBar.SetResizePolicy(ResizePolicy::USE_NATURAL_SIZE, Dimension::HEIGHT);
+progressBar.ValueChangedSignal().Connect( this, &ProgressBarExample::OnValueChanged );
+Stage::GetCurrent().Add( progressBar );
+```
+
+<a name="3"></a>
+## ProgressBar Properties
+
+You can modify the progress bar appearance and behavior through its properties.
+
+To change a property from its default value, use the `SetProperty()` function:
+
+```
+// Create a timer to update the progress of all progress bars
+mTimer = Timer::New( 50 );
+mTimer.TickSignal().Connect( this, &ProgressBarExample::OnTimerTick );
+mTimer.Start();
+
+...
+
+// To set the values, use SetProperty()
+bool OnTimerTick()
+{
+  mProgressValue += 0.01f; // float
+  mSecondaryProgressValue = mProgressValue + 0.1f; // float
+  progressBar.SetProperty( ProgressBar::Property::PROGRESS_VALUE, mProgressValue );
+  progressBar.SetProperty( ProgressBar::Property::SECONDARY_PROGRESS_VALUE, mSecondaryProgressValue );
+
+  // Only call again if progress has NOT got to the end
+  return ( mProgressValue < 1.0f );
+}
+```
+
+The following table lists the available progress bar properties.
+
+**Table: ProgressBar properties**
+
+| Property                   | Type    | Description                                   |
+| -------------------------- | ------- | --------------------------------------------- |
+| `PROGRESS_VALUE`           | FLOAT   | The progress value of progress bar. Runs from 0 to 1. |
+| `SECONDARY_PROGRESS_VALUE` | FLOAT   | The secondary progress value of progress bar. Runs from 0 to 1. |
+| `INDETERMINATE`            | BOOLEAN | Sets the progress bar as indeterminate state. |
+| `TRACK_VISUAL`             | MAP     | The Track Visual value of progress bar. It is a full progress area and shown behind PROGRESS_VISUAL. |
+| `PROGRESS_VISUAL`          | MAP     | The Progress Visual value of progress bar. The size of the progress visual is changed based on PROGRESS_VALUE. |
+| `SECONDARY_PROGRESS_VISUAL` | MAP     | The Secondary Progress Visual of progress bar. The size of the secondary progress visual is changed based on SECONDARY_PROGRESS_VALUE. |
+| `INDETERMINATE_VISUAL`     | MAP     | The indeterminate visual of progress bar.     |
+| `INDETERMINATE_VISUAL_ANIMATION` | MAP or ARRAY | The transition data for indeterminate visual animation. |
+| `LABEL_VISUAL`             | MAP     | The Label visual of progress bar.             |
+
+<a name="4"></a>
+## Customizing the ProgressBar Appearance
+
+The progress bar provides a set of default images, which are used automatically if you do not specify anything else.
+
+If you want to customize the progress bar appearance, you can assign your own images using the `Property::Map` class. You can set the size and image of the track, progress region, secondary progress region, and indeterminate visual.
```

Comment:
Suggested edit = To customize the appearance of the progress bar, you can assign external images using the Property::Map class. This also requires you to set the size of the track, a progress region, a secondary progress region, and an indeterminate visual.

Reply:
Done.